### PR TITLE
Expose Debugger.NotifyOfCrossThreadDependency in contract

### LINKED ISSUE #5868: Debugger.NotifyOfCrossThreadDependency missing from System.Diagnostics.Debug contract
NotifyOfCrossThreadDependency is used in code any time a debugger accessing that property would potentially cause deadlocks, race conditions, or other behavior changes, e.g. if the debugger has frozen all threads in the process (which it does at a breakpoint), and a property getting evaluated blocks waiting for another thread to perform some operation.

We should expose this in the contract.  If the runtime is built for a platform that doesn't support it, it can simply be a nop.

### PR DESCRIPTION
**Pull request by stephentoub**
Goes with:
https://github.com/dotnet/coreclr/pull/3340
https://github.com/dotnet/corert/pull/933

Fixes #5868

### CODE REVIEW

File: src/System.Diagnostics.Debug/ref/System.Diagnostics.Debug.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.11.0</AssemblyVersion>
+    <AssemblyVersion>4.1.0.0</AssemblyVersion>
```
Comment (stephentoub) [inline src/System.Diagnostics.Debug/ref/System.Diagnostics.Debug.csproj]: @weshaggard, is this correct?  I'm assuming I have to bump the version number due to the new method, but it wasn't 100% clear to me.  Are there any other such changes that need to be made that I missed?

Comment (weshaggard) [inline src/System.Diagnostics.Debug/ref/System.Diagnostics.Debug.csproj]: Yes this is correct.

**Comment by weshaggard** [comment]
@stephentoub this change needs to be made in TFS. Currently any changes that bump the versions need to be made in TFS otherwise they will break the build when they mirror in.

**Comment by stephentoub** [comment]
> Currently any changes that bump the versions need to be made in TFS otherwise they will break the build when they mirror in.

Ok. What else needs to be updated as a result of the version number change?

**Comment by stephentoub** [comment]
Closing to make the change internally and let it mirror.

**Pull request closed without merging**